### PR TITLE
add reference to build as a label on built images

### PR DESCRIPTION
**Pull request by bparees**
fixes https://github.com/openshift/origin/issues/5978

### CODE REVIEW

**Comment by bparees** [comment]
@openshift/devex ptal, ultimately this implements the agreement reached here:
https://github.com/openshift/origin/issues/5978#issuecomment-292970237

**Comment by bparees** [comment]
[test]

**Comment by bparees** [comment]
[testextended][extended:core(builds)]

File: pkg/build/builder/util.go [hunk 1]
```diff
@@ -192,3 +194,10 @@ func reportPushFailure(err error, authPresent bool, pushAuthConfig docker.AuthCo
 	}
 	return fmt.Errorf("Failed to push image: %v", err)
 }
+
+// addBuildLabels adds some common image labels describing the build that produced
+// this time.
+func addBuildLabels(labels map[string]string, build *buildapi.Build, labelNamespace string) {
```
Comment (csrwng) [inline pkg/build/builder/util.go]: why pass the labelNamespace? shouldn't we just use the constant?

Comment (bparees) [inline pkg/build/builder/util.go]: done

**Comment by csrwng** [comment]
LGTM

File: pkg/build/builder/util.go [hunk 2]
```diff
@@ -192,3 +194,10 @@ func reportPushFailure(err error, authPresent bool, pushAuthConfig docker.AuthCo
 	}
 	return fmt.Errorf("Failed to push image: %v", err)
 }
+
+// addBuildLabels adds some common image labels describing the build that produced
```
Comment (coreydaley) [inline pkg/build/builder/util.go]: grammar nit: that was produced

Comment (bparees) [inline pkg/build/builder/util.go]: actually it's correct as is.  what's incorrect is "this time" on the next line which should read "this image"

Comment (bparees) [inline pkg/build/builder/util.go]: fixed

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/11114
[merge]

**Pull request merged**